pkg/controller/nodelifecycle/node_lifecycle_controller.go

```diff
@@ -624,12 +624,15 @@ func (nc *Controller) doNoScheduleTaintingPass(nodeName string) error {
 	// Map node's condition to Taints.
 	var taints []v1.Taint
 	for _, condition := range node.Status.Conditions {
-		if taintMap, found := nodeConditionToTaintKeyStatusMap[condition.Type]; found {
-			if taintKey, found := taintMap[condition.Status]; found {
-				taints = append(taints, v1.Taint{
-					Key:    taintKey,
-					Effect: v1.TaintEffectNoSchedule,
-				})
+		// ingnore nodeReady conditions as they're handled by the taint manager
+		if nc.runTaintManager && condition.Type != v1.NodeReady {
```

Comment:
This is confusing.  Is this code not part of the taintManager itself?  If this code is part of taintManager, then saying 'nodeReady conditions handled by taint manager' doesn't make sense.

If this code is not the taint manager, then what should happen if runTaintManager is false?  Who will be setting these taints?

Reply:
I have the same question ^. Also the bug says that Unreachable taint with NoExecute effect is not set for NodeReady=Unknown. With this change even the Unreachable=NoSchedule taint won't get set?